### ISSUE TITLE
Memory keeps increasing and causes app to crash

### ISSUE THREAD

**Issue by kyriakos89** [spritebuilder/SpriteBuilder#812]
Project Link: https://www.dropbox.com/s/2n5y2hfquus1n97/TestMemory.spritebuilder.zip?dl=0

![screen shot 2014-08-30 at 20 07 59](https://cloud.githubusercontent.com/assets/8600099/4099652/ebb930ac-3067-11e4-93da-53c474ec2d1e.png)

CASE 1: 
I have create a fresh project using the latest version of Spritebuilder (v3.2) and placed a single restart button calling:
[[CCDirector sharedDirector]replaceScene:[CCBReader loadAsScene:@"MainScene"]];

Every time a press the restart button, the memory goes up around 0.3mb and eventually causes the app to crash. 

I have also been experiencing memory leaks in several other areas of my projects caused by other functions.

CASE 2:
1) Create a fresh SpriteBuilder project with a CCLabelTTF and set DocRootVar as _text.
2) Declare the following variables in MainScene:
# import "MainScene.h"

@implementation MainScene{
    CCLabelTTF *_text;
    int counter;
    NSString *string;
}
- (void)didLoadFromCCB {
  counter=0;
  }

-(void)update:(CCTime)delta{
    CCLOG(@"Update is working");
    counter++;
    string = [NSString stringWithFormat:@"%d",counter];
    [_text setString:string];
}
All commands in update are required to re-produce the memory leak. If either counter++ or [_text setString:string]; are commented out, the memory leak is gone.

I have attached a project to demonstrate the bug. You can measure memory either inside xCode or instruments using the leak profiling tool.

I have found many other threats that people have been reporting memory leaks but it seems that no one has fixed them. We have been trying spritebuilder for about a week in my company since it looks like a great tool that can earn you precious time , but if this memory issues are not solved we will abandon it. frowning Developers pls fix asap!!!

**Comment by cocojoe** [comment]
Thanks, I tested a few scenarios and agree that there is something going on, wondering if the texture created for the TTF is not being released for whatever reason.  The `Scene` appears to dealloc just fine.

**Comment by kyriakos89** [comment]
@cocojoe  Does the button implementation use a CCLabelTTF?  If you remove the button and call replace scene using a scheduler the leak is gone. So you are right that both issues might be related to TTF not being released for whatever reason.

```
- (void)didLoadFromCCB {
    [self schedule:@selector(restart) interval:1.0f];

}

-(void)restart{
    CCLOG(@"Restart");
    [[CCDirector sharedDirector]replaceScene:[CCBReader loadAsScene:@"MainScene"]];
}
```

**Comment by cocojoe** [comment]
Fixed in latest 3.2, will be part of SB 1.2.1 to be released shortly.
